lib/editors.js

```diff
@@ -401,6 +388,104 @@ class Editor {
             _lock();
         });
     }
+
+    getNamesForCopy(oldShortName, shortNames, oldLongName, longNames) {
```

Comment:
I really like this new code when I played around with it -- it felt very natural.